test/unit/payment_icon_test.rb

```diff
@@ -11,7 +11,16 @@ class PaymentIconTest < ActiveSupport::TestCase
   end
 
   test '#path returns path within app/assets/images' do
-    visa = PaymentIcon.where(name: 'visa').first
-    assert_equal 'payment_icons/visa.svg', visa.path
+    PaymentIcon.all.each do |icon|
+      assert_equal "payment_icons/#{icon.name}.svg", icon.path
+    end
+  end
+
+  test 'Every payment icon record has a corresponding SVG file' do
+    ICONS_DIRECTORY = Rails.root.join("..", "..", "app/assets/images/payment_icons")
```

Comment:
Using `Rails.root.join("..", "..", ...)` to navigate to the actual project root is required since this is a gem/engine. Otherwise, `Rails.root` returns the `dummy` folder from running the test suite.

Reply:
Can you try `PaymentIcons::Engine.root` or `PaymentIcons::Engine.instance.root`?